kong/cluster_events/init.lua

```diff
@@ -133,7 +131,7 @@ function _M.new(opts)
   end
 
   -- set node id (uuid)
-
+  local knode = (kong and kong.node) and kong.node or require "kong.pdk.node".new()
```

Comment:
There is an extra conditional in here that can be removed:

```suggestion
  local knode = kong and kong.node or require "kong.pdk.node".new()
```

```
# when `kong` is nil
$ lua -e 'kong = nil; print(kong and kong.node or "fallback")'
fallback
```
```
# when `kong.node` is nil
$ lua -e 'kong = {}; print(kong and kong.node or "fallback")'
fallback
```
```
# when `kong` and `kong.node` are not nil
$ lua -e 'kong = { node = "kong.node" }; print(kong and kong.node or "fallback")'
kong.node
```

Alternatively, you can preserve the parens to demonstrate explicit precedence (for the sake of the reader) while removing the excess conditional:

```lua
local knode = (kong and kong.node) or require "kong.pdk.node".new()
```

Reply:
@flrgh Thank you for your detailed explanation!

done in this commit: https://github.com/Kong/kong/pull/9067/commits/adfe946c5cb4a24d8f54f68ee89deebda61e9bd9